src/foam/dao/AuthorizationSink.java

```diff
@@ -16,29 +16,29 @@
   extends ProxySink
 {
   public Authorizer authorizer_;
+  public Boolean checkDelete_;
 
   public AuthorizationSink(X x, Authorizer authorizer, Sink delegate) {
+    this(x, authorizer, delegate, false);
+  }
+
+  public AuthorizationSink(X x, Authorizer authorizer, Sink delegate, Boolean checkDelete) {
     super(x, delegate);
     authorizer_ = authorizer;
+    checkDelete_ = checkDelete;
   }
 
   @Override
   public void put(Object obj, Detachable sub) {
     try {
-      authorizer_.authorizeOnRead(getX(), (FObject) obj);
+      if ( checkDelete_ ) {
+        authorizer_.authorizeOnDelete(getX(), (FObject) obj);
```

Comment:
Is updating really the same as deleting?

Reply:
I'm not sure what you mean. I don't see how this relates to updating anything.

Reply:
put() is called to update an object but you're checking if the user is authorized to delete it. I don't think that updating an object is really the same as deleting it outright, which is what the check is checking.

Reply:
I'm assuming either you thought this was a DAO or I don't understand how sink puts work.

> put() is called to update an object

The `put` method defined for DAO's does, but this is a sink, not a DAO. The `put` method for sinks is called when adding something to a sink, sink puts have nothing to do with updating objects, right? Or am I misunderstanding something?

Reply:
I see. My mistake.